src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -56,20 +66,21 @@ ReactRef.shouldUpdateRefs = function(prevElement, nextElement) {
   // is made. It probably belongs where the key checking and
   // instantiateReactComponent is done.
 
-  var prevEmpty = prevElement === null || prevElement === false;
-  var nextEmpty = nextElement === null || nextElement === false;
-
   return (
     // This has a few false positives w/r/t empty components.
-    prevEmpty || nextEmpty ||
+    prevElement == null || prevElement === false || // prevEmpty
```

Comment:
It should be safe. If we got this far then there's no way it could be undefined, as `.ref` access would blow up anyway.

I think by convention rendered nodes can only be `null`, `false`, or `string|number|ReactElement`. Practically you can never get an `undefined` here.

It would be nice if we could enforce this somehow but I guess making the check more relaxed should not be a big deal. We can think about it again if we find more similar checks.